pkgs/development/python-modules/yark/default.nix

```diff
@@ -1,26 +1,52 @@
-{ lib, buildPythonPackage, fetchPypi,
-  click, colorama, flask, requests, yt-dlp }:
+{ lib
+, buildPythonPackage
+, fetchPypi
+, poetry-core
+, click
+, colorama
+, flask
+, requests
+, yt-dlp
+, progress
+}:
 
 buildPythonPackage rec {
   pname = "yark";
-  version = "1.2.3";
+  version = "1.2.4";
+
+  format = "pyproject";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "sha256-KMnQpEH2Z19Y0jBjqx2rZjmlle2M9bcuDCjDIljQEYY=";
+    sha256 = "fdd50d0e832b4522cbe24433f42ed571a1f199f571c1d0c98472b94a501db9cf";
   };
 
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace 'requests = "^2.28.2"' 'requests = "*"'
+  '';
```

Comment:
Could `pythonRelaxDepsHook` work?

Reply:
yes, it works. I didn't know about that hook. Thanks!